File: backend/lib/L1_domain/entities/tracker/person.py
#  Copyright (c) 2022. Alexandr Moroz

from ..base_entity import DBPersistEntity
from .base_tracker import Importable


class Person(DBPersistEntity, Importable):
    firstname: str | None
    lastname: str | None

    # TODO: валидация на пустые поля. Чтобы хоть что-то заполнено было...

File: frontend/lib/L1_domain/entities/goals/person.dart
// Copyright (c) 2022. Alexandr Moroz

import '../base_entity.dart';

class Person extends RPersistable {
  Person({
    required int id,
    required this.workspaceId,
    required this.email,
    this.firstname,
    this.lastname,
  }) : super(id: id);

  final String? firstname;
  final String? lastname;
  final String email;
  final int workspaceId;

  @override
  String toString() => '${firstname ?? ''}${(lastname ?? '').isNotEmpty ? ' ' : ''}${lastname ?? ''}';
}
